### ISSUE TITLE
Allow users to publish directly to a private repository

### ISSUE THREAD

**Issue by dpennell** [scalacenter/sbt-release-early#4]
This is a feature request so that those users that must use private repositories can use them instead of bintray or sonatype.

**Comment by jvican** [comment]
Hello @dpennell, a private repository located where? Bintray? Sonatype?
  
I've just added support for Sonatype. But I'm not planning to work on this. If you want to give it a shot, I'll provide you guidance and help.

**Comment by dpennell** [comment]
Oops,  I missed all of the thread updates..

@jvican in my case, the private repository is an on-prem installation of artifactory.
My focus on this has been preempted for the moment - it will likely be a few weeks before I could work on a PR.

**Comment by jvican** [comment]
Cool @dpennell, I would welcome warmly such a contribution :smile:.
  
Just to give me a little bit more of context: what kind of integration would you need? Isn't it enough to have sbt-sonatype support to publish to artifactory?

**Comment by jvican** [comment]
@dpennell Can you give more context on this ticket? Like, what would need to be added? Doesn't sbt-sonatype allow you to publish packages to your own artifactory repository? It may very well be the case that this feature is already supported, so I wanna find out what you meant.

**Comment by jvican** [comment]
I'm closing @dpennell. Whenever you feel like providing more info, we can reopen!

**Comment by nightscape** [comment]
We also have this scenario. I'm not deeply into either Artifactory or Sonatype, but would assume that they're using quite different protocols for releasing artifacts. Unfortunately I can't test thoroughly now :disappointed: 
Btw, why does `sbt-release-early` not just use the configured `publishTo` and `credentials`?
And would it maybe make sense to add some methods to the [`UnderlyingPublisher` trait](https://github.com/scalacenter/sbt-release-early/blob/d5a820af6c678e6961c7d77f7cecfbb972d4d97a/src/main/scala/ch/epfl/scala/sbt/release/ReleaseEarlyPlugin.scala#L29-L31) and override them in the `case objects`? That way one could create a new `UnderlyingPublisher` for e.g. Artifactory.

**Comment by jvican** [comment]
Absolutely, `UnderlyingPublisher` was meant to do that, and allow for that addition. However, there's an even easier solution: use the default `SonatypePublisher` and override `publishTo` and `credentials`. The rest should work.

**Comment by jvican** [comment]
Well, actually, no -- use `BintrayPublisher` which relies on `publishSigned` to do the right job. The sonatype publisher delegates to the specific sonatype command.
  
If that doesn't work, adding `Artifactory` to `UnderlyingPublisher` should work. Then, modify the parts of the code that assume that there are only two publishers and you're good to go.

**Comment by jvican** [comment]
@nightscape You finally interested in taking a stab at this?
